File: in_Java/0697 Degree of an Array.java
class Solution {
    public int findShortestSubArray(int[] nums) {
        Map<Integer,Integer> hm = new HashMap<>();
        for(int i : nums){
            if(!hm.containsKey(i)) hm.put(i,1);
            else hm.put(i, hm.get(i)+1);
        }
        List<Integer> L = new ArrayList<>();
        int freq = Integer.MIN_VALUE;
        for(int i : hm.values()){
            if(i > freq) freq = i;
        }
        for(int k : hm.keySet()){
            if(hm.get(k) == freq) L.add(k);
        }
        int res = Integer.MAX_VALUE;
        int[] x;
        for(int i : L){
            x = idxOf(nums,i);
            if(res > x[1] - x[0]) res = x[1] - x[0];
        }
        return res+1;
    }
    private int[] idxOf(int[] n, int i){
        int[] res = new int[2];
        int id = 0;
        while(n[id] != i){
            id++;
        }
        res[0] = id;
        int last = n.length-1;
        while(n[last] != i){
            last--;
        }
        res[1] = last;
        return res;
    }
}

File: in_Python/0697 Degree of an Array.py
class Solution(object):
    def findShortestSubArray(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        d = {}
        f = -1
        fm = []
        for i in nums:
            if i not in d:
                d[i] = 1
            else:
                d[i] += 1
            if d[i] > f:
                f = d[i]
        for k in d:
            if d[k] == f:
                fm.append(k)
        res = 50001
        for f in fm:
            i,j = 0,0
            for i in range(len(nums)):
                if nums[i] == f:
                    break
            for j in range(len(nums)-1, -1,-1):
                if nums[j] == f:
                    break
            res = min(res, j-i+1)
        return res